sdk/tables/data-tables/src/TableTransaction.ts

```diff
@@ -111,20 +114,24 @@ export class InternalTableTransaction {
    */
   constructor(
     url: string,
+    tableName: string,
     partitionKey: string,
-    interceptClient: TableClientLike,
-    transactionGuid: string,
-    transactionRequest: InnerTransactionRequest,
+    transactionId: string,
+    changesetId: string,
     credential?: TablesSharedKeyCredentialLike
   ) {
     this.credential = credential;
-    this.partitionKey = partitionKey;
     this.url = url;
-    this.transactionGuid = transactionGuid;
-    this.transactionRequest = transactionRequest;
-    this.pendingOperations = [];
+    this.interceptClient = new TableClient(this.url, tableName);
 
-    this.interceptClient = interceptClient;
+    // Reset-able properties
```

Comment:
Would it be possible for us to just call `reset` here instead of having the code duplicated?

Reply:
I was trying to do that but TS is complaining on lines 99-103 that the properties are not assigned in the constructor. I'm investigating if there is a way to tell TS that we are actually initializing those.

Reply:
You can use a definite assignment assertion when declaring them like this

```ts
private transactionId!: string;
private changesetId!: string;
```

Reply:
Okay, @richardpark-msft gave me the idea of making these reset-able properties an object and having the initialize function return the initialized object. This gives us a nice separation of the properties that may be reset while allowing us for code reuse. Thanks, guys for your help improving this 😄

Reply:
@xirzec I didn't see your comment before my last post. What do you think about having these reset-able properties as an object?

Reply:
I'm fine with that approach